dataprofiler/tests/profilers/test_categorical_column_profile.py

```diff
@@ -936,6 +1140,93 @@ def test_json_decode_after_update(self):
 
         test_utils.assert_profiles_equal(deserialized, expected_profile)
 
+    def test_cms_heavy_hitters_threshold(self):
+
+        df_categorical = pd.Series(
+            [
+                "a",
+                "a",
+                "a",
+                "a",
+                "a",
+                "b",
+                "b",
+                "b",
+                "b",
+                "b",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+            ]
+        )
+        options = CategoricalOptions()
+        options.cms = True
+        options.cms_confidence = 0.95
+        options.cms_relative_error = 0.01
+        options.heavy_hitters_threshold = 2
+
+        profile = CategoricalColumn("test_name", options)
+        profile.update(df_categorical)
+
+        self.assertEqual(1, len(profile.categories))
+        self.assertTrue(profile.sample_size >= 10)
+
+    def test_cms_update_hybrid_batch_stream(self):
+        dataset = pd.Series(["a"] * 4 + ["b"] * 12 + ["c"] * 14)
+        dataset1 = pd.Series(["a"] * 9 + ["b"] * 11 + ["c"] * 9 + ["d"] * 1)
+
+        options = CategoricalOptions()
+        options.cms = True
+        options.cms_confidence = 0.95
+        options.cms_relative_error = 0.01
+        options.heavy_hitters_threshold = 3
+
+        profile = CategoricalColumn("test_name", options)
+        profile.update(dataset)
```

Comment:
we should do an assert after this first.

then a second assert after the next update.

We can then validate the update as expected. probably good to have  the case where a was not initially included, but is after.

Reply:
modified test to address this case